x-pack/plugins/apm/public/components/app/Home/index.tsx

```diff
@@ -83,7 +83,8 @@ interface Props {
 }
 
 export function Home({ tab }: Props) {
-  const { config } = useApmPluginContext();
+  const { config, core } = useApmPluginContext();
+  const isMLEnabled = !!core.application.capabilities.ml;
```

Comment:
if ML is disabled in elasticsearch or disabled in the current space, this check will still be true.
I'd suggest looking for `!!core.application.capabilities.ml?.canAccessML`
`canAccessML` is the very minimum ML capability and is `true` on a basic license, so the current link to the start trial page will still work correctly.